### PR TITLE
Added models : image, analysis, bounding box and brand

### LINKED ISSUE #41: Set up database for data persistance
We want to save analyses to the database for future use. This first ticket deals with setting up the database, so including the relevant tables etc.

Image

The ERD above shows the tables we want to create. The `accounts_user` table has already been created, so we need to create the other four. The naming convention of the tables is such that the model class should be called `BoundingBox` and be inside the `images` app. Django will then give it the name `images_boundingbox`. We also have some useful abstract model classes you can extend to give the models the default columns, such as `pkid`, `id` etc. We will want certain fields to allow null, such as `images_boundingbox.brand`, and `images_image.source` but most we will not want to be null

Note: for `float` fields, it is better to use `models.DecimalField` as it is more accurate
Note: the `brands_brand` table will need to be created in a new app called `brands`

A super important thing is to **not** run `make migrations` until the pull request has been opened and reviewed. So you can create the model classes and the `brands` app, then open the pull request so we can double check everything, then run `make migrations` and commit the migrations files

### PR DESCRIPTION
**Pull request by dhrvrc**

### CODE REVIEW

File: api/images/models/analysis.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from django.db import models
+
+from api.models import BaseModel
+
+
+class Analysis(BaseModel):
+    image = models.ForeignKey("Image", on_delete=models.CASCADE, blank=False, null=False)
+    user = models.ForeignKey("accounts.User", on_delete=models.CASCADE, blank=False, null=False)
```
Comment (sktylr) [inline api/images/models/analysis.py]:
```suggestion
from api.models import BaseUserOwnedModel


class Analysis(BaseUserOwnedModel):
    image = models.ForeignKey("Image", on_delete=models.CASCADE, blank=False, null=False, to_field="id")
```

File: api/images/models/bounding_box.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from django.db import models
+
+from api.models import BaseModel
+
+
+class BoundingBox(BaseModel):
+    image_analysis = models.ForeignKey("Analysis", blank=False, null=False)
```
Comment (sktylr) [inline api/images/models/bounding_box.py]:
```suggestion
    image_analysis = models.ForeignKey("Analysis", blank=False, null=False, to_field="id")
```

File: api/images/models/bounding_box.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    x = models.DecimalField(blank=False, null=False)
+    y = models.DecimalField(blank=False, null=False)
+    width = models.DecimalField(blank=False, null=False)
+    height = models.DecimalField(blank=False, null=False)
+    confidence = models.DecimalField(blank=False, null=False)
```
Comment (sktylr) [inline api/images/models/bounding_box.py]: You can add an arbitrary maximum digit size
```suggestion
    x = models.DecimalField(blank=False, null=False, max_digits=14)
    y = models.DecimalField(blank=False, null=False, max_digits=14)
    width = models.DecimalField(blank=False, null=False, max_digits=14)
    height = models.DecimalField(blank=False, null=False, max_digits=14)
    confidence = models.DecimalField(blank=False, null=False, max_digits=14)
```

File: api/images/models/bounding_box.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    brand = models.ForeignKey("brands.Brand", on_delete=models.CASCADE, blank=False, null=True)
```
Comment (sktylr) [inline api/images/models/bounding_box.py]:
```suggestion
    brand = models.ForeignKey("brands.Brand", on_delete=models.CASCADE, blank=False, null=True, to_field="id")
```

File: api/images/models/bounding_box.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    user = models.ForeignKey("accounts.User", on_delete=models.CASCADE, blank=False, null=False)
```
Comment (sktylr) [inline api/images/models/bounding_box.py]: We can remove this and inherit from `BaseUserOwnedModel` like above

File: api/images/models/image.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+from django.db import models
+
+from api.models import BaseModel
+
+
+class Image(BaseModel):
+    source = models.URLField(blank=False, null=True)
+    user = models.ForeignKey("accounts.User", on_delete=models.CASCADE, blank=False, null=False)
```
Comment (sktylr) [inline api/images/models/image.py]:
```suggestion
from api.models import BaseUserOwnedModel


class Image(BaseUserOwnedModel):
    source = models.URLField(blank=False, null=True)
```

**Review by sktylr — COMMENTED** [review]
Nice work! You'll also want to add the brands app to the `api.settings.INSTALLED_APPS` list and in both models directories (`brands/models`, `images/models`) you'll want to create an `__init__.py` and import all models from that app into it before running `make migrations`

File: api/images/models/image.py (hunk 6 above)
Comment (dhrvrc) [reply api/images/models/image.py]: Made these changes, I just wanted to confirm everything is okay before running `make migrations`

**Review by sktylr — APPROVED** [review]
Awesome that looks all good so you can run `make migrations` and then commit the new migrations files. Let me know if there are any issues with making the migrations, it can sometimes be a little bit funky for the first time

**Review by egaynor — APPROVED** [review]

**Review by sktylr — APPROVED** [review]
🔥

**Pull request merged**